lib/http-bridge/test/integration/Test/Integration/Scenario/CLISpec.hs

```diff
@@ -0,0 +1,144 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Integration.Scenario.CLISpec
+    ( specNoCluster
+    , specWithCluster
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Api.Types
+    ( ApiWallet )
+import Control.Monad
+    ( forM_ )
+import Data.List
+    ( length )
+import Data.Proxy
+    ( Proxy (..) )
+import Data.Text
+    ( Text )
+import System.Command
+    ( Exit (..), Stderr (..), Stdout (..), command )
+import System.Exit
+    ( ExitCode (..) )
+import Test.Hspec
+    ( SpecWith, describe, it )
+import Test.Hspec.Expectations.Lifted
+    ( shouldBe, shouldContain, shouldNotContain )
+import Test.Integration.Framework.DSL
+    ( Context (..)
+    , Headers (..)
+    , Payload (..)
+    , expectResponseCode
+    , expectValidJSON
+    , getFromResponse
+    , json
+    , request
+    , walletId
+    )
+
+import qualified Data.Text as T
+import qualified Network.HTTP.Types.Status as HTTP
+
+specNoCluster :: SpecWith ()
+specNoCluster = do
+    it "CLI - Shows help on bad argument" $  do
+        (Exit c, Stdout out) <- command [] "cardano-wallet" ["--bad arg"]
+        out `shouldContain` "Cardano Wallet CLI"
+        c `shouldBe` ExitFailure 1
+
+    it "CLI - Shows help with --help" $  do
+        (Exit c, Stdout out) <- command [] "cardano-wallet" ["--help"]
+        out `shouldContain` "Cardano Wallet CLI"
+        c `shouldBe` ExitSuccess
+
+    it "CLI - Shows version" $  do
+        (Exit c, Stdout out) <- command [] "cardano-wallet" ["--version"]
+        let v = T.dropWhileEnd (== '\n') (T.pack out)
+        v `shouldBe` "2019.5.8"
+        c `shouldBe` ExitSuccess
+
+    it "CLI - Can generate mnemonics with default size" $  do
+        (Exit c, Stdout out) <- command [] "cardano-wallet"
+            ["mnemonic", "generate"]
+        length (words out) `shouldBe` 15
+        c `shouldBe` ExitSuccess
+
+    it "CLI - It can't generate mnemonics with an invalid size" $ do
+        (Exit c, Stdout out, Stderr err) <- command [] "cardano-wallet"
+            ["mnemonic", "generate", "--size", "14"]
+        c `shouldBe` ExitFailure 1
+        err `shouldBe` "Invalid mnemonic size. Expected one of: 9,12,15,18,21,24\n"
+        out `shouldBe` mempty
+
+    describe "CLI - Can generate mnemonics with different sizes" $ do
+        let test size = it ("--size=" <> show size) $ do
+                (Exit c, Stdout out) <- command [] "cardano-wallet"
+                    ["mnemonic", "generate", "--size", show size]
+                length (words out) `shouldBe` size
+                c `shouldBe` ExitSuccess
+        forM_ [9, 12, 15, 18, 21, 24] test
+
+specWithCluster :: SpecWith Context
+specWithCluster = do
+    it "CLI - Can get a wallet" $ \ctx -> do
+        walId <- createWallet ctx "1st CLI Wallet" mnemonics15
```

Comment:
why not use arbitrary `name` and `mnemonics`? (I can help with this if needed)

Reply:
¯\_(ツ)_/¯

Reply:
One _benefit_ in case of `mnemonics` is that they were all generated with https://iancoleman.io/bip39/ (i.e. using something that is outside of our codebase, which makes sense in terms of testing/verifying that they work)